src/container.cpp

```diff
@@ -550,26 +548,26 @@ void Container::updateThing(Thing* thing, uint16_t itemId, uint32_t count)
 	}
 }
 
-void Container::replaceThing(uint32_t index, Thing* thing)
+void Container::replaceThing(uint32_t index, std::shared_ptr<Thing> thing)
 {
 	if (!thing) {
 		return;
 	}
 
-	Item* item = thing->getItem();
+	auto item = thing->getItem();
 	if (!item) {
 		return /*RETURNVALUE_NOTPOSSIBLE*/;
 	}
 
-	Item* replacedItem = getItemByIndex(index);
+	auto replacedItem = getItemByIndex(index);
```

Comment:
Could this be `const auto`
```suggestion
	const auto replacedItem = getItemByIndex(index);
```